realm/realm-library/src/androidTestObjectServer/java/io/realm/SyncedRealmMigrationTests.java

```diff
@@ -51,6 +51,14 @@
     @Rule
     public final TestSyncConfigurationFactory configFactory = new TestSyncConfigurationFactory();
 
+    @BeforeClass
+    public static void beforeClass () {
+        // another Test class may have the BaseRealm.applicationContext set but
+        // the SyncManager reset. This will make assertion to fail, we need to re-initialise
+        // the sync_manager.cpp#m_file_manager (configFactory rule do this)
+        BaseRealm.applicationContext = null;
```

Comment:
do I need to do that if I want to log a user out? :smile:

Reply:
no, this is just to reset an overhead from other tests where the ObjectStore SyncManager is reset. Setting `BaseRealm.applicationContext` to `null` here will just force another `ObjectServer#init`